### PR TITLE
Made the cache factory strict

### PR DESCRIPTION
**Pull request by rullzer**
* Return types
* Typehints
* made strict

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

### CODE REVIEW

File: lib/private/Memcache/Factory.php [hunk 1]
```diff
@@ -31,8 +31,10 @@
 
 namespace OC\Memcache;
 
+use OCP\ICache;
 use \OCP\ICacheFactory;
```
Comment (ChristophWurst) [inline lib/private/Memcache/Factory.php]: I'd prefer consistency here 😉

Comment (rullzer) [reply lib/private/Memcache/Factory.php]: fair enough... I gues one was hand writen and the other a phpstorm import ;)

**Review by juliusknorr — APPROVED** [review]

**Review by ChristophWurst — APPROVED** [review]

**Comment by MorrisJobke** [comment]
ref #7392

**Comment by MorrisJobke** [comment]
Unit tests fail

**Pull request merged**